dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryAPIImpl.java

```diff
@@ -978,4 +973,17 @@ public PaginatedCategories findAll(final CategorySearchCriteria searchCriteria,
 		return getCategoriesSubList(searchCriteria.offset, searchCriteria.limit, categories, null);
 	}
 
+	/**
+	 * Default implementation of {@link CategoryAPI}
+	 *
+	 * @param inodes List of inodes to search
+	 * @return
+	 *
+	 * @throws DotDataException
+	 */
+	@CloseDBIfOpened
+	@Override
+	public List<HierarchyShortCategory> findHierarchy(final Collection<String> inodes) throws DotDataException {
+		return categoryFactory.findHierarchy(inodes);

```

Comment:
what about permissions?

Reply:
We need to discuss about this, and maybe create another issue